### PR TITLE
Mode improvements for UI

### PR DESCRIPTION
**Pull request by gavofyork**
Addressing #3049.

### CODE REVIEW

File: parity/user_defaults.rs [hunk 1]
```diff
@@ -67,11 +85,28 @@ impl Visitor for UserDefaultsVisitor {
 		let fat_db: Value = map.remove("fat_db".into()).unwrap_or_else(|| Value::Bool(false));
 		let fat_db = try!(fat_db.as_bool().ok_or_else(|| Error::custom("invalid fat_db value")));
 
+		let mode: Value = map.remove("mode".into()).unwrap_or_else(|| Value::String("active".to_owned()));
+		let mode = match try!(mode.as_str().ok_or_else(|| Error::custom("invalid mode value"))) {
+			"off" => Mode::Off,
```
Comment (arkpar) [inline parity/user_defaults.rs]: "offline" probably?

File: parity/helpers.rs [hunk 2]
```diff
@@ -58,8 +58,8 @@ pub fn to_mode(s: &str, timeout: u64, alarm: u64) -> Result<Mode, String> {
 		"active" => Ok(Mode::Active),
 		"passive" => Ok(Mode::Passive(Duration::from_secs(timeout), Duration::from_secs(alarm))),
 		"dark" => Ok(Mode::Dark(Duration::from_secs(timeout))),
-		"off" => Ok(Mode::Off),
-		_ => Err(format!("{}: Invalid address for --mode. Must be one of active, passive or dark.", s)),
+		"offline" => Ok(Mode::Off),
+		_ => Err(format!("{}: Invalid value for --mode. Must be one of active, passive or dark.", s)),
```
Comment (rphmeier) [inline parity/helpers.rs]: ...or offline?

File: ethcore/src/client/client.rs [hunk 3]
```diff
@@ -866,7 +873,14 @@ impl BlockChainClient for Client {
 	fn mode(&self) -> IpcMode { self.mode.lock().clone().into() }
 
 	fn set_mode(&self, mode: IpcMode) {
-		*self.mode.lock() = mode.clone().into();
+		{
+			let mut mode = self.mode.lock();
+			*mode = mode.clone().into();
+			match *self.on_mode_change.lock() {
+				Some(ref mut f) => f(&*mode),
+				_ => {}
```
Comment (rphmeier) [inline ethcore/src/client/client.rs]: `if let Some`

File: mac/post-install.sh [hunk 4]
```diff
@@ -26,9 +24,11 @@ cat > $HOME/Library/LaunchAgents/io.parity.ethereum.plist <<EOF
 </dict>
 </plist>
 EOF
-mkdir -p $HOME/.parity
 
-chown $USER $HOME/.parity $HOME/Library/LaunchAgents $HOME/Library/LaunchAgents/io.parity.ethereum.plist
+mkdir -p $HOME/.parity/906a34e69aec8c0d
+echo -n '{"fat_db":false,"mode":"passive","pruning":"archive","tracing":false}' > $HOME/.parity/906a34e69aec8c0d/user_defaults
```
Comment (arkpar) [inline mac/post-install.sh]: "pruning":"archive" ?

**Pull request merged**